addons/website_event_crm/models/event_registration.py

```diff
@@ -26,6 +26,36 @@ def _get_lead_description_fields(self):
         res.append('registration_answer_ids')
         return res
 
+    def _get_lead_grouping(self, rules, rule_to_new_regs):
+        visitor_registrations = self.filtered('visitor_id')
+        # We don't want to group by visitor if there is a sale order, this is done here to avoid
+        # another bridge module.
+        if self._fields.get('sale_order_id'):
+            visitor_registrations -= visitor_registrations.filtered('sale_order_id')
+        grouping_res = super(EventRegistration, self - visitor_registrations)._get_lead_grouping(rules, rule_to_new_regs)
+
+        if visitor_registrations:
+            related_registrations = self.env['event.registration'].search([
+                ('visitor_id', 'in', visitor_registrations.visitor_id.ids)
+            ])
+            related_leads = self.env['crm.lead'].search([
+                ('event_lead_rule_id', 'in', rules.ids),
+                ('registration_ids', 'in', related_registrations.ids),
+            ])
+
+            for rule in rules:
+                rule_new_regs = rule_to_new_regs[rule]
+                visitor_to_regs = (rule_new_regs & visitor_registrations).grouped('visitor_id')
+                visitor_res = []
+                for visitor, registrations in visitor_to_regs.items():
+                    registrations = registrations.sorted('id')
+                    leads = related_leads.filtered(lambda lead: lead.event_lead_rule_id == rule and lead.registration_ids.visitor_id == visitor)
```

Comment:
lead.registration_ids.visitor_id : I always have a doubt. This returns a singleton, right? The first visitor_id. Could we have more than one visitor on registrations linked to a lead? :exploding_head: